tools/build-kani/src/sysroot.rs

```diff
@@ -106,7 +106,12 @@ pub fn build_lib() {
 
     // Collect the build artifacts.
     let artifacts = build_artifacts(&mut cmd);
-    let _ = cmd.wait().expect("Couldn't get cargo's exit status");
+    let exit_status = cmd.wait().expect("Couldn't get cargo's exit status");
+    // exit_ok is an experimental API where we could do .exit_ok().expect("Build failed")
+    // Until it is stable:
+    if !exit_status.success() {
+        panic!("Build failed");
```

Comment:
IMHO, the current usage of `expect` is for very unusual situations (I/O related), e.g. failing to spawn a process, failing to wait for a child process, failing to create a directory, etc. The case of a non-zero exit status for `cargo build` is not as unusual as it could happen due to an error in the Kani library during development. Thus returning an error here seems more appropriate.

Reply:
Fair point, this now actually makes for a much more reasonable behaviour (such as no longer suggesting the use of RUST_BACKTRACE for debugging...). Fixed, but I'd appreciate you taking another look whether this is the most appropriate way to do it.